src/main/java/org/sciborgs1155/robot/drive/Drive.java

```diff
@@ -1,263 +1,181 @@
+  @Log private final SimpleMotorFeedforward RdriveFF = new SimpleMotorFeedforward(DriveConstants.kS, DriveConstants.kV, DriveConstants.kA);
+  @Log private final SimpleMotorFeedforward LdriveFF = new SimpleMotorFeedforward(DriveConstants.kS, DriveConstants.kV, DriveConstants.kA);
 
-  @Log private double speedMultiplier = 1;
+  
 
-  public static Drive create() {
-    return Robot.isReal()
-        ? new Drive(
-            new MAXSwerveModule(
-                MODULE_NAMES[0], FRONT_LEFT_DRIVE, FRONT_LEFT_TURNING, ANGULAR_OFFSETS[0]),
-            new MAXSwerveModule(
-                MODULE_NAMES[1], FRONT_RIGHT_DRIVE, FRONT_RIGHT_TURNING, ANGULAR_OFFSETS[1]),
-            new MAXSwerveModule(
-                MODULE_NAMES[2], REAR_LEFT_DRIVE, REAR_LEFT_TURNING, ANGULAR_OFFSETS[2]),
-            new MAXSwerveModule(
-                MODULE_NAMES[3], REAR_RIGHT_DRIVE, REAR_RIGHT_TURNING, ANGULAR_OFFSETS[3]))
-        : new Drive(new SimModule(), new SimModule(), new SimModule(), new SimModule());
-  }
+  /**
+   * Encoders PID and FF controllers, -> mostly done by now, maybe there is something else about this that can be worked on later
+   * DifferentialDriveOdometry -> used DifferentialDrivePoseEstimator instead rn
+   * Getters for pose, encoder values
+   * Setters for setpoints and other important values
+   * Style
+   */
+  
 
-  public Drive(ModuleIO frontLeft, ModuleIO frontRight, ModuleIO rearLeft, ModuleIO rearRight) {
-    this.frontLeft = frontLeft;
-    this.frontRight = frontRight;
-    this.rearLeft = rearLeft;
-    this.rearRight = rearRight;
+  //sets up Odometry
+  private final DifferentialDriveKinematics kinematics = new DifferentialDriveKinematics(ROBOT_TRACK);
+  DifferentialDriveWheelSpeeds speeds = new DifferentialDriveWheelSpeeds(0, 0);
 
-    modules = List.of(frontLeft, frontRight, rearLeft, rearRight);
-    modules2d = new FieldObject2d[modules.size()];
+  @Log private final DifferentialDrivePoseEstimator odometry = 
+    new DifferentialDrivePoseEstimator(
+      kinematics,
+      gyro.getRotation2d(), 
+      leftEncoder.getPosition(), 
+      rightEncoder.getPosition(), 
+      new Pose2d(),
+          VecBuilder.fill(0.05, 0.05, Units.degreesToRadians(5)), //these could probably be adjusted later, taken from wpilib docs
+          VecBuilder.fill(0.5, 0.5, Units.degreesToRadians(30)));
 
-    odometry =
-        new SwerveDrivePoseEstimator(kinematics, getHeading(), getModulePositions(), new Pose2d());
+  /** Creates a new Drive. */
+  public Drive(){
+    leftEncoder.setPositionConversionFactor(DriveConstants.GEAR_RATIO * DriveConstants.WHEEL_CIRCUMFERENCE);
+    rightEncoder.setPositionConversionFactor(DriveConstants.GEAR_RATIO * DriveConstants.WHEEL_CIRCUMFERENCE);
 
-    for (int i = 0; i < modules2d.length; i++) {
-      modules2d[i] = field2d.getObject("module-" + i);
-    }
-  }
+    leftEncoder.setVelocityConversionFactor(DriveConstants.GEAR_RATIO);
```

Comment:
Velocity conversion factor should be set to the exact same value as position conversion fact, times 60.
This is because the native units for position are 1 = 1 rotation (your position conversion factor converts this into meters) and the native units for velocity are 1 = 1 rotation / minute. We deal with velocity in meters / second.